src/solvers/strings/string_concatenation_builtin_function.cpp

```diff
@@ -1,17 +1,32 @@
 /*******************************************************************\
 
-Module: Generates string constraints for functions adding content
-        add the end of strings
+Module: Builtin functions for string concatenations
 
-Author: Romain Brenguier, romain.brenguier@diffblue.com
+Author: Romain Brenguier, Joel Allred
 
 \*******************************************************************/
 
 /// \file
-/// Generates string constraints for functions adding content add the end of
-///   strings
+/// Builtin functions for string concatenations
 
-#include "string_constraint_generator.h"
+#include "string_concatenation_builtin_function.h"
+
+#include <algorithm>
+
+string_concatenation_builtin_functiont::string_concatenation_builtin_functiont(
```

Comment:
:bulb: Why not move this constructor below the free functions and just above the methods of this class?

Reply:
I discussed with Romain and we think it's a bit more welcoming to start with the constructor, since it doesn't depend on the other functions.